kernel/byterun/coq_gc.h

```diff
@@ -1,59 +0,0 @@
-/***********************************************************************/
-/*                                                                     */
-/*                           Coq Compiler                              */
-/*                                                                     */
-/*        Benjamin Gregoire, projets Logical and Cristal               */
-/*                        INRIA Rocquencourt                           */
-/*                                                                     */
-/*                                                                     */
-/***********************************************************************/
-
-#ifndef _COQ_CAML_GC_
-#define _COQ_CAML_GC_
-#include <caml/mlvalues.h>
-#include <caml/alloc.h>
-#include <caml/memory.h>
-
-typedef void (*scanning_action) (value, value *);
-
-
-CAMLextern char *young_ptr;
-CAMLextern char *young_limit;
-CAMLextern void (*scan_roots_hook) (scanning_action);
-CAMLextern void minor_collection (void);
```

Comment:
It took me a while to guess where this was declared now, until I remembered that C allows calls to undeclared functions :thinking: 
It is declared in caml/minor_gc.h only since 4.10, so please declare it again in the branch for OCaml<4.10.

Reply:
C99 doesn't allow such calls, only C90 and previous standards do. (See https://en.wikipedia.org/wiki/ANSI_C#C99. GCC and Clang are a bit lax with enforcing that, but you can make them enforce it with `-Werror=implicit-function-declaration`.)